lib/recovery.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Microsemi Switchtec(tm) PCIe Management Library
+ * Copyright (c) 2017, Microsemi Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/**
+ * @file
+ * @brief Switchtec core library functions for secure boot operations
+ */
+
+#include "switchtec_priv.h"
+#include "switchtec/switchtec.h"
+#include "switchtec/errors.h"
+#include "switchtec/endian.h"
+#include "switchtec/mrpc.h"
+#include "switchtec/errors.h"
+#include <unistd.h>
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+
+#define IS_UNSUPPORTED_MRPC(x) ((x & (~SWITCHTEC_ERRNO_MRPC_FLAG_BIT)) == \
+				ERR_MPRC_UNSUPPORTED)
```

Comment:
I agree with Logan that we have a common macro like ERRNO_ MRPC() to get error, with which we can also test other errors. And it seems it's a common macro which better be placed in a .h file.